### PR TITLE
Allow KCFinder to send back a json encoded response instead of string…

### PR DESCRIPTION
**Pull request by seamuslee001**
… javascript

ping @colemanw

### CODE REVIEW

File: kcfinder/core/class/uploader.php [hunk 1]
```diff
@@ -759,11 +768,19 @@ protected function callBack($url, $message="") {
                 $js = $this->$method($url, $message);
         }
 
-        if (!isset($js))
+        if ($this->outputFormat == 'json') {
+          $json = $this->callBack_json($url, $message);
+          echo json_encode($json);
```
[the remark this answers is not in the document]
Comment (seamuslee001) [reply kcfinder/core/class/uploader.php]: @colemanw what would you recommend for content type?

File: kcfinder/core/class/uploader.php [hunk 2]
```diff
@@ -759,11 +768,20 @@ protected function callBack($url, $message="") {
...
+          $js = $this->callBack_ckeditor($url, $message);
+          if (!isset($js)) {
             $js = $this->callBack_default($url, $message);
 
-        header("Content-Type: text/html; charset={$this->charset}");
-        echo "<html><body>$js</body></html>";
+            header("Content-Type: text/html; charset={$this->charset}");
+            echo "<html><body>$js</body></html>";
+          }
+        }
```
Comment (colemanw) [inline kcfinder/core/class/uploader.php]: There's something odd going on here. The deleted line on 762 has no opening bracket, but the line you aded on 778 does. I think that may be causing the odd behavior.

File: kcfinder/core/class/uploader.php [hunk 3]
```diff
@@ -759,11 +768,19 @@ protected function callBack($url, $message="") {
...
+          echo json_encode($json);
+        }
+        else {
+          if (!isset($js)) {
             $js = $this->callBack_default($url, $message);
 
-        header("Content-Type: text/html; charset={$this->charset}");
-        echo "<html><body>$js</body></html>";
+            header("Content-Type: text/html; charset={$this->charset}");
+            echo "<html><body>$js</body></html>";
```
Comment (colemanw) [inline kcfinder/core/class/uploader.php]: The above 2 lines should not be inside that conditional. The original version of this file only has 1 line "inside" the conditional (which is what happens when an if statement doesn't have brackets.

**Comment by seamuslee001** [comment]
thanks @colemanw should be fixed now me thinks

**Comment by colemanw** [comment]
This looks good now and is working correctly according to my testing.

**Pull request merged**